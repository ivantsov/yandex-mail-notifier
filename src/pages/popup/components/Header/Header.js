import React, {PropTypes} from 'react';
import i18n from 'shared/utils/i18n';
import SettingsIcon from './Icons/SettingsIcon';
import ReloadIcon from './Icons/ReloadIcon';
import ComposeIcon from './Icons/ComposeIcon';

import styles from './Header.less';

const Header = ({
    user,
    unreadMessagesCount,
    disabled,
    reloadMessages,
    openLink,
    openSettings,
}) => (
    <div className={styles.component}>
        <a className={styles.composeBtn} onClick={() => openLink('#compose')}>
            <ComposeIcon className={styles.composeIcon}/>
            {i18n.text('popup.compose')}
        </a>
        <div className={styles.centerBlock}>
            <a onClick={() => openLink()}>{/* we need callback here, otherwise event will be passed as a first param*/}
                {user} (<strong>{unreadMessagesCount}</strong>)
            </a>
            <button
                className={styles.reloadBtn}
                disabled={disabled}
                onClick={() => reloadMessages(true)}
            >
                <ReloadIcon className={styles.reloadIcon}/>
            </button>
        </div>
        <a className={styles.item} onClick={openSettings}>
            <SettingsIcon className={styles.settingsIcon}/>
        </a>
    </div>
);

Header.propTypes = {
    user: PropTypes.string.isRequired,
    unreadMessagesCount: PropTypes.number.isRequired,
    disabled: PropTypes.bool.isRequired,
    reloadMessages: PropTypes.func.isRequired,
    openLink: PropTypes.func.isRequired,
    openSettings: PropTypes.func.isRequired,
};

export default Header;
