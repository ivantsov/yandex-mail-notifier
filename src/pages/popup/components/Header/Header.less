@import "../../styles/colors";

.component {
    background-color: white;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 8px 10px;
    border-bottom: 4px solid @yellow1;
}

.composeBtn {
    display: flex;
    align-items: center;
}

.composeIcon {
    width: 16px;
    height: 16px;
    margin-right: 8px;
}

.centerBlock {
    display: inline-flex;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
}

.reloadIcon {
    width: 16px;
    height: 16px;
    margin-left: 8px;
    fill: @green;
    transition: 0.3s transform;

    &:hover {
        transform: rotate(180deg);
    }
}

.reloadBtn {
    width: 16px;
    height: 16px;
    margin-left: 10px;
}

.settingsIcon {
    width: 18px;
    height: 18px;
    fill: @blue1;
    transition: 0.4s transform;

    &:hover {
        transform: rotate(300deg);
    }
}
