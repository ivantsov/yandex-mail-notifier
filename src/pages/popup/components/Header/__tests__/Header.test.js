import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

const reloadMessages = jest.fn();
const openLink = jest.fn();
const openSettings = jest.fn();

const commonProps = {
    user: 'username@ya.ru',
    unreadMessagesCount: 5,
    disabled: false,
    reloadMessages,
    openLink,
    openSettings,
};

function render(props) {
    const component = renderer.create(
        <Header
            {...commonProps}
            {...props}
        />,
    );

    return component.toJSON();
}

describe('popup/Header', () => {
    beforeEach(() => {
        window.chrome = {
            i18n: {
                getMessage: jest.fn(() => 'text'),
            },
        };
    });

    describe('render', () => {
        it('disabled', () => {
            const tree = render({disabled: true});
            expect(tree).toMatchSnapshot();
        });

        it('active', () => {
            const tree = render();
            expect(tree).toMatchSnapshot();
        });
    });

    it('callbacks', () => {
        const tree = render();
        const [
            composeLink,
            centerBlock,
            settingsLink,
        ] = tree.children;

        composeLink.props.onClick();
        expect(openLink).lastCalledWith('#compose');

        const [mailLink, reloadBtn] = centerBlock.children;

        mailLink.props.onClick();
        expect(openLink).lastCalledWith();

        reloadBtn.props.onClick();
        expect(reloadMessages).lastCalledWith();

        settingsLink.props.onClick();
        expect(openSettings).lastCalledWith();
    });
});
