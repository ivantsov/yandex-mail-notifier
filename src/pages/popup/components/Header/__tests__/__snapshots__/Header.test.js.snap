// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup/Header render active 1`] = `
<div
  className="component"
>
  <button
    className="primary"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="composeBtn"
    >
      <svg
        className="composeIcon"
        viewBox="0 0 16 16"
      >
        <path
          d="M14 0L6 8l-1 3 3-1 8-8-2-2zM0 2v14h14v-6h-2v4H2V4h3V2H0z"
        />
      </svg>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "text",
          }
        }
      />
    </span>
  </button>
  <div
    className="centerBlock"
  >
    <a
      onClick={[Function]}
    >
      username@ya.ru
       (
      <strong>
        5
      </strong>
      )
    </a>
    <button
      className="icon"
      disabled={false}
      onClick={[Function]}
    >
      <svg
        className="reloadIcon"
        viewBox="0 0 16 16"
      >
        <path
          d="M8 3c1.202 0 2.29.44 3.152 1.148l-2.154 1.97L16 7.845 14.9.717l-2.25 2.06C11.41 1.675 9.786 1 8 1 5.398 1 3.133 2.423 1.926 4.53l1.75 1C4.54 4.025 6.143 3 8 3zm0 9.98c-1.202 0-2.29-.44-3.152-1.148l2.154-1.97L0 8.135l1.1 7.127 2.25-2.06C4.59 14.305 6.214 14.98 8 14.98c2.602 0 4.867-1.423 6.074-3.53l-1.75-1C11.46 11.955 9.857 12.98 8 12.98z"
        />
      </svg>
    </button>
  </div>
  <button
    className="icon"
    disabled={false}
    onClick={[Function]}
  >
    <svg
      className="settingsIcon"
      viewBox="1.5 2 20 19"
    >
      <path
        d="M11.4,15.4c2.2,0,4-1.8,4-4s-1.8-4-4-4s-4,1.8-4,4S9.2,15.4,11.4,15.4z M14.9,2l3,1.7l-0.7,2.8l0.3,0.4c0.2,0.3,0.4,0.5,0.6,0.6c0.2,0.3,0.4,0.5,0.5,0.7l0.2,0.5l2.7,0.7v3.8l-2.7,0.6l-0.3,0.6c-0.3,0.6-0.6,1.2-1,1.8l-0.3,0.4l0.8,2.7L14.9,21l-1.5-1.7L13,19h-0.5c-0.7,0.1-1.3,0.1-2,0H9.9l-0.4,0.4L8.1,21l-2.9-1.8l0.7-2.7l-0.3-0.4c-0.2-0.3-0.5-0.5-0.7-0.8c-0.2-0.2-0.4-0.3-0.5-0.5l-0.2-0.5l-2.7-0.7l0.1-4l2.2-0.5L4.4,9l0.2-0.5C4.8,8,5.1,7.4,5.5,6.9l0.3-0.4L4.9,3.8L8.2,2l1.5,1.7l0.4,0.4L10.6,4c0.6-0.1,1.2-0.1,1.9,0L13,4.1L14.9,2z"
      />
    </svg>
  </button>
</div>
`;

exports[`popup/Header render disabled 1`] = `
<div
  className="component"
>
  <button
    className="primary"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="composeBtn"
    >
      <svg
        className="composeIcon"
        viewBox="0 0 16 16"
      >
        <path
          d="M14 0L6 8l-1 3 3-1 8-8-2-2zM0 2v14h14v-6h-2v4H2V4h3V2H0z"
        />
      </svg>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "text",
          }
        }
      />
    </span>
  </button>
  <div
    className="centerBlock"
  >
    <a
      onClick={[Function]}
    >
      username@ya.ru
       (
      <strong>
        5
      </strong>
      )
    </a>
    <button
      className="icon"
      disabled={true}
      onClick={[Function]}
    >
      <svg
        className="reloadIcon"
        viewBox="0 0 16 16"
      >
        <path
          d="M8 3c1.202 0 2.29.44 3.152 1.148l-2.154 1.97L16 7.845 14.9.717l-2.25 2.06C11.41 1.675 9.786 1 8 1 5.398 1 3.133 2.423 1.926 4.53l1.75 1C4.54 4.025 6.143 3 8 3zm0 9.98c-1.202 0-2.29-.44-3.152-1.148l2.154-1.97L0 8.135l1.1 7.127 2.25-2.06C4.59 14.305 6.214 14.98 8 14.98c2.602 0 4.867-1.423 6.074-3.53l-1.75-1C11.46 11.955 9.857 12.98 8 12.98z"
        />
      </svg>
    </button>
  </div>
  <button
    className="icon"
    disabled={false}
    onClick={[Function]}
  >
    <svg
      className="settingsIcon"
      viewBox="1.5 2 20 19"
    >
      <path
        d="M11.4,15.4c2.2,0,4-1.8,4-4s-1.8-4-4-4s-4,1.8-4,4S9.2,15.4,11.4,15.4z M14.9,2l3,1.7l-0.7,2.8l0.3,0.4c0.2,0.3,0.4,0.5,0.6,0.6c0.2,0.3,0.4,0.5,0.5,0.7l0.2,0.5l2.7,0.7v3.8l-2.7,0.6l-0.3,0.6c-0.3,0.6-0.6,1.2-1,1.8l-0.3,0.4l0.8,2.7L14.9,21l-1.5-1.7L13,19h-0.5c-0.7,0.1-1.3,0.1-2,0H9.9l-0.4,0.4L8.1,21l-2.9-1.8l0.7-2.7l-0.3-0.4c-0.2-0.3-0.5-0.5-0.7-0.8c-0.2-0.2-0.4-0.3-0.5-0.5l-0.2-0.5l-2.7-0.7l0.1-4l2.2-0.5L4.4,9l0.2-0.5C4.8,8,5.1,7.4,5.5,6.9l0.3-0.4L4.9,3.8L8.2,2l1.5,1.7l0.4,0.4L10.6,4c0.6-0.1,1.2-0.1,1.9,0L13,4.1L14.9,2z"
      />
    </svg>
  </button>
</div>
`;
